File: exam/collaguazo/u1/exam/src/ec/edu/espe/model/Notebook/Notebook.java
package ec.edu.espe.model.Notebook;
/**
 *
 * @author Pablo Collaguazo
 */
public class Notebook {
    private String id;
    private String brand;
    private int pages;
    private String size;
    private double price;
    
    // Constructor por defecto
    public Notebook() {}
    
    // Constructor con parámetros
    public Notebook(String id, String brand, int pages, String size, double price) {
        this.id = id;
        this.brand = brand;
        this.pages = pages;
        this.size = size;
        this.price = price;
    }
    
    // Getters y Setters
    public String getId() { return id; }
    public void setId(String id) { this.id = id; }
    
    public String getBrand() { return brand; }
    public void setBrand(String brand) { this.brand = brand; }
    
    public int getPages() { return pages; }
    public void setPages(int pages) { this.pages = pages; }
    
    public String getSize() { return size; }
    public void setSize(String size) { this.size = size; }
    
    public double getPrice() { return price; }
    public void setPrice(double price) { this.price = price; }
    
    @Override
    public String toString() {
        return String.format("Notebook [ID: %s, Brand: %s, Pages: %d, Size: %s, Price: $%.2f]", 
                           id, brand, pages, size, price);
    }
    
    // Método para convertir a JSON
    public String toJson() {
        return String.format(
            "{\"id\":\"%s\",\"brand\":\"%s\",\"pages\":%d,\"size\":\"%s\",\"price\":%.2f}",
            id, brand, pages, size, price
        );
    }
}

File: hw/collaguazo/u3/hw26GUIs/Noteboo_Manager/models/notebook.py
class Notebook:
    def __init__(self, id, brand, pages, size, price):
        self.id = id
        self.brand = brand
        self.pages = pages
        self.size = size
        self.price = price
    
    def to_dict(self):
        return {
            'id': self.id,
            'brand': self.brand,
            'pages': self.pages,
            'size': self.size,
            'price': self.price
        }
    
    @classmethod
    def from_dict(cls, data):
        return cls(
            id=data['id'],
            brand=data['brand'],
            pages=data['pages'],
            size=data['size'],
            price=data['price']
        )
    
    def __str__(self):
        return f"Notebook [ID: {self.id}, Brand: {self.brand}, Pages: {self.pages}, Size: {self.size}, Price: ${self.price:.2f}]"